src/command_executor.cc

```diff
@@ -624,7 +624,7 @@ execute_file(exec_context& ec, const std::string& path_and_args, bool multiline)
 
     vars["#"] = env_arg_name;
     for (size_t lpc = 0; lpc < split_args.size(); lpc++) {
-        snprintf(env_arg_name, sizeof(env_arg_name), "%lu", lpc);
+        snprintf(env_arg_name, sizeof(env_arg_name), "%lu", (long unsigned int)lpc);
```

Comment:
I believe there's a `z` length modifier, so this should be `%zu` and then you don't need to cast it.

Reply:
You are correct -- removing the suggested cast and using "z" format works as well for this case.